File: solutions/0001-0100/0061/0061.js
var rotateRight = function(head, k) {
  if (!head) return head;

  let len = 1;
  let tail = head;
  let newHead = tail;

  while (tail.next) {
    tail = tail.next;
    len++;
  }
  tail.next = head;

  if (k %= len) {
    for (let i = 0; i < len - k; i++) tail = tail.next;
  }

  newHead = tail.next;
  tail.next = null;

  return newHead;
};

File: solutions/0001-0100/0061/0061.py
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
      if not head: return head

      len = 1;
      newHead = tail = head;

      while tail.next:
        tail = tail.next
        len += 1
      tail.next = head

      k = k % len
      if k:
        for i in range(len - k): tail = tail.next

      newHead = tail.next
      tail.next = None

      return newHead